### PR TITLE
Alignment of variable names in general.

### PR DESCRIPTION
**Pull request by zionyx**
* Renamed `-Image` → `-ImageName`.
* Updated all tests to accommodate the change.
* Removed `-Repository`.
* Fixed broken test cases of `Invoke-DockerPull`.
* Added more tests to improve test coverage.

### CODE REVIEW

File: Public/Invoke-DockerPull.ps1 [hunk 1]
```diff
@@ -23,18 +23,18 @@ function Invoke-DockerPull {
...
 
-    $registryPostfixed = Add-Postfix -Registry $Registry
+    $postfixedRegistry = Add-Postfix -Value $Registry
 
     # Pulls by tag by default
-    $imageToPull = "${registryPostfixed}${Image}:${Tag}"
+    $imageToPull = "${postfixedRegistry}${ImageName}:${Tag}"
 
-    # Digest overrides tag however
+    # Digest cannot be used together with Tag
```
Comment (rasmusjelsgaard) [inline Public/Invoke-DockerPull.ps1]: much better 🌞

File: Public/Invoke-DockerTag.ps1 [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
         [Parameter(mandatory=$true)]
         [String]
-        $TargetImage,
+        $NewImageName,
 
         [ValidateNotNullOrEmpty()]
         [String]
-        $TargetTag = 'latest'
+        $NewTag = 'latest'
     )
```
Comment (rasmusjelsgaard) [inline Public/Invoke-DockerTag.ps1]: Would it make sense to check if $NewTag -eq $Tag to be able to return immediately if that's the case?

File: Tests/Find-ImageName.Tests.ps1 [hunk 3]
```diff
@@ -33,6 +33,13 @@ Describe 'Parse context from git repository' {
             $result | Should -BeExactly "dockerbuild-pwsh"
         }
 
+        It 'can identify an invalid repository, .git folder without config file' {
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]: Lovely 🌞

File: Tests/Invoke-DockerTag.Tests.ps1 [hunk 4]
```diff
@@ -24,91 +24,91 @@ Describe 'Tag docker images' {
...
             $result = GetMockValue -Key "mock"
             Write-Debug $result
-            $result | Should -BeLikeExactly "docker tag lalaland:latest lololand:latest"
+            $result | Should -BeLikeExactly "docker tag oldname:latest newimage:latest"
         }
 
-        It 'tags docker image with image name with pester tag' {
-            Invoke-DockerTag -SourceImage 'lalaland' -SourceTag 'pester' -TargetImage 'lololand' -TargetTag 'pester'
+        It 'tags image by image name and tag with new image name and tag' {
+            Invoke-DockerTag -ImageName 'oldname' -Tag 'pester' -NewImageName 'newimage' -NewTag 'pester'
```
Comment (rasmusjelsgaard) [inline Tests/Invoke-DockerTag.Tests.ps1]: Shouldn't this be 'OldTag' to follow the naming scheme?

File: Tests/Utilities.Tests.ps1 [hunk 5]
```diff
@@ -30,20 +30,30 @@ Describe 'Validate various support functions for testing' {
             Test-Path -Path (Join-Path $pathInTemp ".git") -PathType Container | Should -Be $true
             Test-Path -Path (Join-Path $pathInTemp ".git/config") -PathType Leaf | Should -Be $true
 
-            Remove-Item $pathInTemp -Recurse -Force | Out-Null
+            Remove-Item $pathInTemp -Recurse -Force
```
Comment (rasmusjelsgaard) [inline Tests/Utilities.Tests.ps1]: Out of curiosity, why did you remove the Out-Null?

File: Public/Invoke-DockerTag.ps1 (hunk 2 above)
Comment (zionyx) [reply Public/Invoke-DockerTag.ps1]: What do you mean by return immediately?
Like do nothing when both `Tag` and `NewTag` have the same value?
If that's the case, then the comparison will involve comparing if `$source -eq $target`, where `source/target = ${Registry}${ImageName}${Tag}`.

File: Tests/Invoke-DockerTag.Tests.ps1 (hunk 4 above)
Comment (zionyx) [reply Tests/Invoke-DockerTag.Tests.ps1]: Yes, that thought crossed my mind. But to stay consistent with the rest of the cmdlets, i chose to reuse the same variable names for original variable type.

File: Tests/Utilities.Tests.ps1 (hunk 5 above)
Comment (zionyx) [reply Tests/Utilities.Tests.ps1]: Not sure. Since it does not show any output in non-debug modes. I removed it to see if it makes any difference. I cross checked with other tests and they have `Out-Null`. So only this file is affected. Just to spark conversations. 😄
